### PR TITLE
yaff

### PR DESCRIPTION
**Pull request by jan-janssen**
@conda-forge/help-python-c

Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your
      recipe (see
      [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos)
      for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by jan-janssen** [comment]
For now it is only the linux version for python 3.7. Still it would be great if somebody could merge this PR, to make yaff available via conda-forge.

File: recipes/molmod/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    - cython >=0.24.1
+    - numpy >=1.0
+    - future
+    - setuptools
+    - gpaw
+  run:
+    - numpy >=1.0
+    - future
+    - cython >=0.24.1
+    - setuptools
```
Comment (isuruf) [inline recipes/molmod/meta.yaml]: Why do you need `cython` and `setuptools` at runtime?

File: recipes/yaff/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  skip: True  # [not linux or py<37]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+  host:
+    - python
+    - pip
+    - numpy>=1.5
+    - cython>=0.26
```
Comment (isuruf) [inline recipes/yaff/meta.yaml]: You need a space before `>=` like `cython >=0.26`

File: recipes/yaff/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  script: "{{ PYTHON }} -m pip install . -vv"
+  skip: True  # [not linux or py<37]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+  host:
+    - python
+    - pip
+    - numpy>=1.5
```
Comment (isuruf) [inline recipes/yaff/meta.yaml]: See https://conda-forge.org/docs/maintainer/knowledge_base.html#building-against-numpy on how to correctly add numpy dependency

File: recipes/molmod/LICENSE.txt [hunk 4]
```diff
@@ -0,0 +1,674 @@
+                    GNU GENERAL PUBLIC LICENSE
```
Comment (isuruf) [inline recipes/molmod/LICENSE.txt]: This file is unnecessary as conda-build will use the one from the source tarball.

File: recipes/yaff/meta.yaml (hunk 3 above)
Comment (isuruf) [reply recipes/yaff/meta.yaml]: Can you please do the same for the other recipe in this PR?

File: recipes/yaff/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    - {{ compiler('c') }}
+  host:
+    - python
+    - pip
+    - numpy
+    - cython >=0.26
+    - matplotlib >1.0.0
+    - h5py >=2.0.0
+    - molmod >=1.4.1
+    - scipy >=0.17.1
```
Comment (isuruf) [inline recipes/yaff/meta.yaml]: Above 4 dependencies are not needed in `host`

File: recipes/molmod/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    - {{ compiler('c') }}
+    - cython >=0.24.1
+  host:
+    - python
+    - pip
+    - cython >=0.24.1
+    - numpy >=1.0
+    - future
+    - setuptools
+    - gpaw
```
Comment (isuruf) [inline recipes/molmod/meta.yaml]: Above 3 requirements are unneeded in host

File: recipes/molmod/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+  run:
+    - numpy >=1.0
+    - future
+    - gpaw
```
Comment (isuruf) [inline recipes/molmod/meta.yaml]: `gpaw` is not a dependency, https://github.com/molmod/molmod/blob/master/setup.py#L96-L97

**Comment by jan-janssen** [comment]
Remove MolMod - wait for https://github.com/conda-forge/staged-recipes/pull/8894 to be merged first

File: recipes/yaff/meta.yaml (hunk 3 above)
Comment (jan-janssen) [reply recipes/yaff/meta.yaml]: I did it in https://github.com/conda-forge/staged-recipes/pull/8894

File: recipes/molmod/meta.yaml (hunk 7 above)
Comment (jan-janssen) [reply recipes/molmod/meta.yaml]: It is not listed but it does not work without it - you can try - I basically got the information from private communication with the developers.

File: recipes/yaff/meta.yaml (hunk 5 above)
Comment (jan-janssen) [reply recipes/yaff/meta.yaml]: But they are listed in the `install_requires`: https://github.com/molmod/yaff/blob/master/setup.py

Comment (isuruf) [reply recipes/yaff/meta.yaml]: You only need the `setup_requires` dependencies in `host` and `install_requires` dependencies are needed in `run`.

File: recipes/yaff/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+  script: "{{ PYTHON }} -m pip install . -vv"
+  skip: True  # [not linux or py<37]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+  host:
+    - python
+    - pip
+    - numpy >=1.11.3
```
Comment (isuruf) [inline recipes/yaff/meta.yaml]: This is incorrect. You need to remove the version inequality or make it a equality. (For equality, you need to use something newer than `1.14`

File: recipes/yaff/meta.yaml (hunk 5 above)
Comment (jan-janssen) [reply recipes/yaff/meta.yaml]: Thx

**Comment by jan-janssen** [comment]
@isuruf I included your feedback, can we merge it now?

**Comment by isuruf** [comment]
Can you open an issue upstream about missing license file?

**Comment by jan-janssen** [comment]
As it is GPL, I thought COPYRIGHT is the right filename, isn't it? The copyright file is also included in the main repository https://github.com/molmod/yaff

**Comment by isuruf** [comment]
Issue is that it is not in the sdist at https://files.pythonhosted.org/packages/3f/21/6269a78287d1b3efce078e0f81566a9e5d2106655a35b83cc2b35eab69e7/yaff-1.4.2.tar.gz

To fix it, a MANIFEST.in needs to be there. See https://github.com/pypa/sampleproject/blob/master/MANIFEST.in#L5

**Pull request merged**